hca/util/__init__.py

```diff
@@ -0,0 +1,332 @@
+"""
```

Comment:
This module contains the core of the entire CLI, yet has barebones documentation.  I don't think this is in a state where other people would be able to come in and be able to work on this codebase.

Reply:
Great point, sorry about that! I've added a docstring, please take a look.

Reply:
The added documentation is a great step, but mostly for someone who wants to understand what this module does.  It's not really that useful for understanding _how_ it works.

If I were to come in and try to work on this, I'd probably have to drop `print(..)` statements everywhere to understand that.